pennylane/ops/op_math/controlled.py

```diff
@@ -764,6 +764,9 @@ def _decompose_custom_ops(op: Controlled) -> List["operation.Operator"]:
         # has some special case handling of its own for further decomposition
         return _decompose_pauli_x_based_no_control_values(op)
 
+    if isinstance(op.base, qml.GlobalPhase) and len(op.control_wires) == 1:
```

Comment:
Should we also add in support for a multi-control global phase? Should just be a controlled phase shift?

Reply:
Unfortunately, I think is not that straightforward. This trick of phase shift only works for a single control Global Phase. If we wanted to do it for $m \geq 2$ controlled qubits, we would have to iteratively decompose the m-controlled gate into a (m-1)-controlled gate where we can apply the trick. And then repeat until m = 1.

While I think is doable, this might make up for a separate story. What do you think?

Reply:
And we don't need to immediately decompose all the way to two qubit gates.  We can always leave it as an n-controlled phase shift until some later point.